### PR TITLE
Avoid unnecessary logs from user injector

### PR DESCRIPTION
**Pull request by ggezer**

### CODE REVIEW

**Comment by ggezer** [comment]
@bf8086 PTAL. Thanks.

File: core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java [hunk 1]
```diff
@@ -102,13 +104,31 @@ public void onHalfClose() {
...
+   * This is mostly to avoid dumping "already closed" exceptions to logs.
+   *
+   * @param call call to close
+   * @param status status to close the call with
+   * @param headers headers to close the call with
+   */
+  private <ReqT, RespT> void safeClose(ServerCall<ReqT, RespT> call, Status status,
+      Metadata headers) {
+    try {
+      call.close(status, headers);
```
Comment (bf8086) [inline core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java]: Add a debug logging message for the authentication error on the server side?

Comment (ggezer) [reply core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java]: Could you clarify your ask? (This is server side)

Comment (ggezer) [reply core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java]: If that's what you means; "Status: {}" in the below debug print will contain with what the call was being closed for.

Comment (bf8086) [reply core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java]: Right now the authentication error message is only reported to the client side through the Status object. We should consider adding debug logging on the server side so even if we don't have the client side message we still have a way to know what went wrong with authentication. This should be logged regardless whether the close of the call result in an exception or not.

Comment (ggezer) [reply core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java]: Got it. Added trace before closing the call.

**Comment by ggezer** [comment]
Jenkins, test this please

File: core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java [hunk 2]
```diff
@@ -102,13 +104,33 @@ public void onHalfClose() {
...
+
+  /**
+   * Closes the call while blanketing runtime exceptions.
+   * This is mostly to avoid dumping "already closed" exceptions to logs.
+   *
+   * @param call call to close
+   * @param status status to close the call with
+   * @param headers headers to close the call with
+   */
+  private <ReqT, RespT> void safeClose(ServerCall<ReqT, RespT> call, Status status,
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java]: Usually this is called `closeQuietly`?

Comment (ggezer) [reply core/common/src/main/java/alluxio/security/authentication/AuthenticatedUserInjector.java]: Sounds better.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**